pymongo/server.py

```diff
@@ -13,22 +13,42 @@
 # permissions and limitations under the License.
 
 """Communicate with one MongoDB server in a topology."""
+from __future__ import annotations
 
 from datetime import datetime
+from typing import TYPE_CHECKING, Any, Callable, List, Mapping, Optional, Tuple, Union
 
 from bson import _decode_all_selective
 from pymongo.errors import NotPrimaryError, OperationFailure
 from pymongo.helpers import _check_command_response, _handle_reauth
-from pymongo.message import _convert_exception, _OpMsg
+from pymongo.message import _convert_exception, _GetMore, _OpMsg, _Query
 from pymongo.response import PinnedResponse, Response
 
+if TYPE_CHECKING:
+    from contextlib import _GeneratorContextManager
+    from queue import Queue
+    from weakref import ReferenceType
+
+    from bson.objectid import ObjectId
+    from pymongo.mongo_client import _MongoClientErrorHandler
+    from pymongo.monitor import Monitor
+    from pymongo.monitoring import _EventListeners
+    from pymongo.pool import Pool, SocketInfo
+    from pymongo.server_description import ServerDescription
+
 _CURSOR_DOC_FIELDS = {"cursor": {"firstBatch": 1, "nextBatch": 1}}
 
 
 class Server:
     def __init__(
-        self, server_description, pool, monitor, topology_id=None, listeners=None, events=None
-    ):
+        self,
+        server_description: ServerDescription,
+        pool: Pool,
+        monitor: Monitor,
+        topology_id: Optional[ObjectId] = None,
+        listeners: Optional[_EventListeners] = None,
+        events: Optional[ReferenceType[Optional[Queue]]] = None,
```

Comment:
Should this be: `Optional[ReferenceType[Queue]]`?

Reply:
When its `Optional[ReferenceType[Queue]]`, I get an incompatible type elsewhere...so I think it does needs to be `Optional[ReferenceType[Optional[Queue]]]`